src/main/kotlin/com/demonwav/mcdev/i18n/inspections/TranslationInspection.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.i18n.inspections
+
+import com.intellij.codeInspection.BaseJavaLocalInspectionTool
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.JavaElementVisitor
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiElementVisitor
+import com.intellij.psi.PsiField
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiMethod
+
+abstract class TranslationInspection : BaseJavaLocalInspectionTool() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return object : JavaElementVisitor() {
+            override fun visitMethod(method: PsiMethod) {
+            }
+
+            override fun visitClass(aClass: PsiClass) {
+            }
+
+            override fun visitField(field: PsiField) {
+            }
+
+            override fun visitElement(element: PsiElement) {
+                checkElement(element, holder)
```

Comment:
There are tons of `PsiElement`s in each Java source files so it would be faster if you had an own `JavaElementVisitor` in each of your inspections (or extract it into another base class) that only runs checks on the elements it needs. For example, in `NoTranslationInspection` you'd only override `visitLiteralExpression`. 

A common pattern I use (and @DemonWav seems to have adopted it for the new sided checks) is to define a named `Visitor` inner classes, hand it the `ProblemsHolder` and then implement the needed `visit` methods in there as needed. Like here: [ShadowTargetInspection.kt](https://github.com/minecraft-dev/MinecraftDev/blob/dev/src/main/kotlin/com/demonwav/mcdev/platform/mixin/inspection/shadow/ShadowTargetInspection.kt)

If you have a lot of inspection which check similar elements you may want to use an extra base class similar to this one: [MixinAnnotationAttributeInspection.kt](https://github.com/minecraft-dev/MinecraftDev/blob/dev/src/main/kotlin/com/demonwav/mcdev/platform/mixin/inspection/MixinAnnotationAttributeInspection.kt) (you can use an anonymous class if you prefer, no difference)